daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env sh
+
+# Valid actions:
+
+# - FATAL       - netdata exited due to a fatal condition
+#      ACTION_RESULT  -- program name and thread tag
+#      ACTION_DATA    -- fmt, args passed to fatal
+# - START       - netdata started
+#      ACTION_DATA     -- nan
+# - EXIT     - installation action
+#      ACTION_DATA    -- ret value of
+
+
+ACTION="${1}"
+ACTION_RESULT="${2}"
+ACTION_DATA="${3}"
+
+# -------------------------------------------------------------------------------------------------
+# check opt-out
+
+if [ -z "${NETDATA_CONFIG_DIR}" ]; then
+    for dir in "$(netdata -W get global 'config directory' '/etc/netdata' 2>/dev/null)" "/etc/netdata" "/opt/netdata/etc/netdata"; do
+        if [ -f "${dir}/.opt-out-from-anonymous-statistics" ]; then
+            exit 0
+        fi
+    done
+else
+    if [ -f "${NETDATA_CONFIG_DIR}/.opt-out-from-anonymous-statistics" ]; then
+        exit 0
+    fi
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the operating system
+
+OS_DETECTION="unknown"
+NAME="unknown"
+VERSION="unknown"
+VERSION_ID="unknown"
+ID="unknown"
+ID_LIKE="unknown"
+
+if [ -f "/etc/os-release" ]; then
+    OS_DETECTION="/etc/os-release"
+    eval "$(grep -E "^(NAME|ID|ID_LIKE|VERSION|VERSION_ID)=" </etc/os-release)"
+elif [ -f "/etc/lsb-release" ]; then
+    OS_DETECTION="/etc/lsb-release"
+    DISTRIB_ID="unknown"
+    DISTRIB_RELEASE="unknown"
+    DISTRIB_CODENAME="unknown"
+    eval "$(grep -E "^(DISTRIB_ID|DISTRIB_RELEASE|DISTRIB_CODENAME)=" </etc/lsb-release)"
+    NAME="${DISTRIB_ID}"
+    VERSION="${DISTRIB_RELEASE}"
+    ID="${DISTRIB_CODENAME}"
+elif [ ! -z "$(command -v lsb_release 2>/dev/null)" ]; then
+    OS_DETECTION="lsb_release"
+    NAME="$(lsb_release -i 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
```

Comment:
`lsb_release -i` should probably be assigned to `ID` and not `NAME`. `NAME` should probably be taken from `lsb_release -d`.

Reply:
Done